subprojects/core/src/main/java/org/gradle/execution/plan/ExecutionNodeAccessHierarchy.java

```diff
@@ -242,12 +259,12 @@ public String getName() {
 
         @Override
         public String getPath() {
-            return relativePath;
+            return relativePath.getPathString();
         }
 
         @Override
         public RelativePath getRelativePath() {
-            return RelativePath.parse(isFile, relativePath);
+            return relativePath;
         }
```

Comment:
These changes don't have a negative impact on performance?

Reply:
In what sense? Because we need to check more things? These changes don't have a negative impact on our performance tests, because I don't think we use filtered file trees as inputs anywhere.

They will make it more expensive to check whether there are missing task dependencies with filtered file trees, though only when the roots of filtered file tree inputs overlap with outputs in the first place. Though these changes are necessary to make the check correct.

Reply:
We are now not creating a new object when you call `getRelativePath()` but we are creating an object when you call `getPath()`. This could have an impact, I was wondering if you saw one, or if there was a reason why there shouldn't be. It could be a positive one, or an "it depends"-kind of impact, too.

Reply:
If you use a pattern like `include/blah/**`, then this pattern will always use `getRelativePath()` for matching and not `getPath()`, so this should actually become faster. If you are using a custom matcher relying on `getPath()`, then it will become slower. I expect most folks to use a pattern, so this should become faster for most folks. Though we don't have any performance tests for this use-case, so all of this is just a guess.